bindings/pyroot_experimental/PyROOT/src/TTreePyz.cxx

```diff
@@ -236,3 +236,155 @@ PyObject *PyROOT::SetBranchAddressPyz(PyObject * /* self */, PyObject *args)
+                        &CPyCppyy_PyUnicode_Type, &clName,
+                        &address,
+                        &PyInt_Type, &bufsize,
+                        &PyInt_Type, &splitlevel)) {
+      bIsMatch = true;
+   } else {
+      PyErr_Clear();
+      if (PyArg_ParseTuple(args, const_cast<char *>("OO!O|O!O!"),
+                           &treeObj,
+                           &CPyCppyy_PyUnicode_Type, &name,
+                           &address,
+                           &PyInt_Type, &bufsize,
+                           &PyInt_Type, &splitlevel)) {
+         bIsMatch = true;
+      } else
+         PyErr_Clear();
+   }
+
+   if (bIsMatch) {
+      auto treeProxy = (CPPInstance *)treeObj;
+      TTree *tree = (TTree *)GetClass(treeProxy)->DynamicCast(TTree::Class(), treeProxy->GetObject());
+      if (!tree) {
+         PyErr_SetString(PyExc_TypeError, "TTree::Branch must be called with a TTree instance as first argument");
+         return nullptr;
+      }
+
+      std::string klName = clName ? CPyCppyy_PyUnicode_AsString(clName) : "";
+      void *buf = nullptr;
+
+      if (CPPInstance_Check(address)) {
+         if (((CPPInstance *)address)->fFlags & CPPInstance::kIsReference)
+            buf = (void *)((CPPInstance *)address)->fObject;
+         else
+            buf = (void *)&((CPPInstance *)address)->fObject;
+
+         if (!clName) {
+            klName = GetClass((CPPInstance *)address)->GetName();
+            argc += 1;
+         }
+      } else
+         Utility::GetBuffer(address, '*', 1, buf, false);
+
+      if (buf && !klName.empty()) {
+         TBranch *branch = 0;
+         if (argc == 4) {
+            branch = tree->Branch(CPyCppyy_PyUnicode_AsString(name), klName.c_str(), buf);
+         } else if (argc == 5) {
+            branch = tree->Branch(CPyCppyy_PyUnicode_AsString(name), klName.c_str(), buf, PyInt_AS_LONG(bufsize));
+         } else if (argc == 6) {
+            branch = tree->Branch(CPyCppyy_PyUnicode_AsString(name), klName.c_str(), buf, PyInt_AS_LONG(bufsize),
+                                  PyInt_AS_LONG(splitlevel));
+         }
+
+         return BindCppObject(branch, Cppyy::GetScope("TBranch"));
+      }
+   }
+
+   Py_RETURN_NONE;
+}
+
+////////////////////////////////////////////////////////////////////////////
+/// \brief Add pythonization for TTree::Branch.
+/// \param[in] self Always null, since this is a module function.
+/// \param[in] args Pointer to a Python tuple object containing the arguments
+/// received from Python.
+///
+/// Modify the behaviour of Branch so that proxy references can be passed
+/// as arguments from the Python side, more precisely in cases where the C++
+/// implementation of the method expects the address of a pointer.
+///
+/// For example:
+/// ~~~{.python}
+/// v = ROOT.std.vector('int')()
+/// t.Branch('my_vector_branch', v)
+/// ~~~
+PyObject *PyROOT::BranchPyz(PyObject * /* self */, PyObject *args)
+{
+   // Acceptable signatures:
```

Comment:
perhaps this is better to put in the doxy?

Reply:
Yes, definitely